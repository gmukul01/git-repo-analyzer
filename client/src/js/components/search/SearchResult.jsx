import React, { PureComponent, Fragment } from "react";
import ReactLoading from "react-loading";

export default class extends PureComponent {
  constructor(props) {
    super(props);
    this.handleClick = this.handleClick.bind(this);
  }

  handleClick(repo) {}

  render() {
    const { isLoading, errorMessage, result } = this.props;
    if (isLoading) {
      return <ReactLoading type="spinningBubbles" color="black" />;
    }

    if (errorMessage) {
      return <p>{`Error : ${errorMessage}`}</p>;
    }

    if (!result.items) {
      return null;
    }

    const rows = result.items.map(repo => (
      <tr key={repo.id} onClick={() => this.handleClick(repo)}>
        <td>{repo.owner.login}</td>
        <td>{repo.name}</td>
        <td>
          <a href={repo.url} target="_blank">
            Link
          </a>
        </td>
      </tr>
    ));

    return (
      <section className="search--table container">
        <table className="table table-hover table-responsive">
          <thead>
            <tr>
              <th>Owner Name</th>
              <th>Repo Name</th>
              <th>Repo Link</th>
            </tr>
          </thead>
          <tbody>{rows}</tbody>
        </table>
      </section>
    );
  }
}
