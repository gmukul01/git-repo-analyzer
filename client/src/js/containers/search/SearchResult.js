import { connect } from "react-redux";

import SearchForm from "components/search/Searchresult";

const mapStateToProps = state => ({
  isLoading: state.search.isLoading,
  errorMessage: state.search.errorMessage,
  result: state.search.result
});

export default connect(
  mapStateToProps,
  null
)(SearchForm);
