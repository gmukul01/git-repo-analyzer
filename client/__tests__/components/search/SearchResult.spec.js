import React from "react";
import { shallow } from "enzyme";
import ShallowRenderer from "react-test-renderer/shallow";

import SearchResult from "components/search/SearchResult";

describe("SearchResult Component", () => {
  const initialProps = {
    result: {
      items: [
        {
          id: "dummyId1",
          owner: { login: "dummyUserName" },
          openPullRequenCount: 1,
          name: "dummyRepoName1",
          url: "dummyUrl1"
        },
        {
          id: "dummyId2",
          owner: { login: "dummyUserName2" },
          openPullRequenCount: 2,
          name: "dummyRepoName2",
          url: "dummyUrl2"
        }
      ]
    }
  };

  it("should match snapshot", () => {
    const renderer = new ShallowRenderer(),
      result = renderer.render(<SearchResult {...initialProps} />);
    expect(result).toMatchSnapshot();
  });

  it("should call addToSearchHistory action on click of any repo", () => {
    const props = { ...initialProps, addToSearchHistory: jest.fn() },
      wrapper = shallow(<SearchResult {...props} />),
      evnt = { preventDefault: jest.fn() };
    wrapper.findWhere(node => node.key() === initialProps.result.items[0].id).simulate("click", evnt, initialProps.result.items[0]);
    expect(props.addToSearchHistory.mock.calls.length).toEqual(1);
  });
});
