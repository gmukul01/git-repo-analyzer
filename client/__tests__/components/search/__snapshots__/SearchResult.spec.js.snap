// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SearchResult Component should match snapshot 1`] = `
<section
  className="table-responsive search-result-table"
>
  <table
    className="table table-hover"
  >
    <thead>
      <tr>
        <th>
          Owner Name
        </th>
        <th>
          Repo Name
        </th>
        <th>
          Repo Link
        </th>
      </tr>
    </thead>
    <tbody>
      <tr
        onClick={[Function]}
      >
        <td>
          dummyUserName
        </td>
        <td>
          dummyRepoName1
        </td>
        <td>
          <a
            href="dummyUrl1"
            target="_blank"
          >
            Link
          </a>
        </td>
      </tr>
      <tr
        onClick={[Function]}
      >
        <td>
          dummyUserName2
        </td>
        <td>
          dummyRepoName2
        </td>
        <td>
          <a
            href="dummyUrl2"
            target="_blank"
          >
            Link
          </a>
        </td>
      </tr>
    </tbody>
  </table>
</section>
`;
